File: tasks/implementations/cpp/tasks/file_io_large.cpp
#include <fstream>
#include <iostream>
#include <string>

void run_file_io_large(const std::string& size, const std::string& fixturesRoot) {
    std::ifstream file(fixturesRoot + "/generated/file_io/" + size + ".txt");
    long long total = 0;
    long long value = 0;
    while (file >> value) total += value;
    std::cout << total << std::endl;
}

File: tasks/implementations/go/tasks/file_io_large.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "path/filepath"
    "strconv"
)

func RunFileIOLarge(size string, fixturesRoot string) {
    file, _ := os.Open(filepath.Join(fixturesRoot, "generated", "file_io", size+".txt"))
    defer file.Close()
    scanner := bufio.NewScanner(file)
    var total int64 = 0
    for scanner.Scan() {
        value, _ := strconv.ParseInt(scanner.Text(), 10, 64)
        total += value
    }
    fmt.Println(total)
}
